classes/spec/decorator_spec.rb

```diff
@@ -0,0 +1,13 @@
+require_relative 'spec_helper'
+
+describe Decorator do
+  before :each do
+    @decorator = Decorator.new('Covey')
+  end
+
+  context 'correct_name' do
+    it 'Check if nameable exists' do
+      expect(@decorator.nameable).to eq 'Covey'
+    end
+  end
+end
```

Comment:
- [ ] Kindly test the `correct_name` method of the `Decorator` class. Remember that all methods in a class should have unit tests for them to ensure that the methods work as expected. This will help you spot bugs in time if there are issues with the method implementation. 👍 😎